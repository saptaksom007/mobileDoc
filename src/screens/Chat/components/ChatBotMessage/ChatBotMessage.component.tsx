import React, { Component } from 'react'
import {
  View,
  ActivityIndicator,
  StyleSheet,
  TouchableOpacity,
} from 'react-native'

import { Icon } from 'react-native-elements'
import {
  Heading3,
  Bold,
  Paragraph,
} from 'components/CustomText/CustomText.component'

import { ChatPositionType } from 'api/types'

import styles, { DEFAULT_COLOR } from './ChatBotMessage.styles'
import { Color } from 'constants/Color'

interface Item {
  title?: string
  subtitle: string
  onPress?(): void
}

interface Props {
  items?: Item[]
  title: string
  iconName: string
  iconType: string
  iconContainerStyle?: any
  position: ChatPositionType
}

class ChatBotMessage extends Component<Props, { isLoading: boolean }> {
  state = {
    isLoading: false,
  }

  static defaultPropTypes = {
    position: 'right',
  }

  handleOnPress = (item: Item) => () => {
    const { onPress } = item
    if (onPress) {
      onPress()
      this.setState(({ isLoading }) => ({ isLoading: !isLoading }))
      setTimeout(() => this.setState({ isLoading: false }), 3000)
    }
  }

  render() {
    const {
      title,
      items,
      iconName,
      iconType,
      position,
      iconContainerStyle,
    } = this.props
    return (
      <>
        <View style={styles.container}>
          {iconName && (
            <View
              style={[
                styles.iconContainer,
                iconContainerStyle,
                position === 'right' && styles.rightIcon,
              ]}
            >
              <Icon name={iconName} type={iconType} color={DEFAULT_COLOR} />
            </View>
          )}
          <View
            style={[
              styles.titleContainer,
              position === 'right' && styles.rightTitleContainer,
            ]}
          >
            <Heading3 style={styles.title}>{title}</Heading3>
          </View>
          {items &&
            items.map((item, index) => (
              <View
                key={index}
                style={[
                  styles.itemContainer,
                  index === items.length - 1 && styles.lastItemContainer,
                ]}
              >
                {!!item.title && (
                  <Bold style={styles.itemTitle}>{item.title || ''}</Bold>
                )}
                <View style={{ flexDirection: 'row' }}>
                  <Paragraph
                    onPress={this.handleOnPress(item)}
                    style={[
                      styles.itemValue,
                      item.onPress && { textDecorationLine: 'underline' },
                    ]}
                  >
                    {item.subtitle
                      ? `${item.subtitle
                          .substr(0, 1)
                          .toUpperCase()}${item.subtitle.substr(1)}`
                      : '[no item subtitle]'}{' '}
                  </Paragraph>
                  {item.onPress && (
                    <TouchableOpacity
                      onPress={this.handleOnPress(item)}
                      hitSlop={{ top: 15, bottom: 5, right: 15 }}
                    >
                      <Icon
                        size={20}
                        name={'open-in-new'}
                        type={'material'}
                        color={DEFAULT_COLOR}
                      />
                    </TouchableOpacity>
                  )}
                </View>
              </View>
            ))}
        </View>
        {this.state.isLoading && (
          <View
            style={[
              StyleSheet.absoluteFill,
              {
                backgroundColor: Color.blackTTTT,
                justifyContent: 'center',
                alignItems: 'center',
                borderRadius: 13,
                opacity: 0.9,
              },
            ]}
          >
            <ActivityIndicator size='small' color={Color.white} />
          </View>
        )}
      </>
    )
  }
}

export default ChatBotMessage
