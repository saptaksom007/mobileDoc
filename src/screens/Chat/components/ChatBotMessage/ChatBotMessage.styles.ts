import { StyleSheet } from 'react-native'
import { Layout } from 'constants/Layout'
import { fontFamilyBold as fontFamily } from 'components/CustomText/CustomText.styles'

export const DEFAULT_COLOR = 'rgba(95, 95, 95, 0.87)'
export const DEFAULT_BACKGROUND = 'rgb(218, 239, 255)'
export const DEFAULT_WIDTH = Layout.window.width / 1.4
export const ICON_SIZE = 30
export const DEFAULT_BORDER_RADIUS = 5

export default StyleSheet.create({
  container: {
    width: DEFAULT_WIDTH,
    backgroundColor: 'transparent',
    borderRadius: 10,
  },
  iconContainer: {
    width: ICON_SIZE,
    height: ICON_SIZE,
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: DEFAULT_BORDER_RADIUS,
    borderColor: '#fff',
    borderWidth: 2,
    top: 10,
    left: -1,
    zIndex: 1,
    backgroundColor: DEFAULT_BACKGROUND,
  },
  titleContainer: {
    backgroundColor: DEFAULT_BACKGROUND,
    marginVertical: 1,
    paddingVertical: 12,
    paddingHorizontal: 7,
    borderTopRightRadius: DEFAULT_BORDER_RADIUS,
  },
  rightTitleContainer: {
    borderTopRightRadius: 0,
    borderTopLeftRadius: DEFAULT_BORDER_RADIUS,
  },
  title: {
    textAlign: 'left',
    color: DEFAULT_COLOR,
    fontFamily,
  },
  itemContainer: {
    backgroundColor: DEFAULT_BACKGROUND,
    marginBottom: 1,
    padding: 7,
  },
  lastItemContainer: {
    borderBottomLeftRadius: DEFAULT_BORDER_RADIUS,
    borderBottomRightRadius: DEFAULT_BORDER_RADIUS,
  },
  itemTitle: {
    color: DEFAULT_COLOR,
    fontSize: 13,
  },
  itemValue: {
    paddingTop: 4,
    paddingBottom: 2,
    color: DEFAULT_COLOR,
  },
  rightIcon: {
    left: DEFAULT_WIDTH - ICON_SIZE + 1,
  },
})
