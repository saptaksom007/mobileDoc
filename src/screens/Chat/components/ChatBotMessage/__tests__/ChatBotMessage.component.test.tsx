import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import ChatBotMessage from '../ChatBotMessage.component'

it('<ChatBotMessage/> render correctly', () => {
  const tree = renderer
    .create(
      <ChatBotMessage
        title="title test"
        items={[{ title: '', subtitle: '' }]}
        iconName="test"
        iconType="test"
        position="right"
      />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
