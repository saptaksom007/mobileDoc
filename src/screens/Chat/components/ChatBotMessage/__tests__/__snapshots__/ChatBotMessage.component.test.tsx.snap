// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChatBotMessage/> render correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "transparent",
      "borderRadius": 10,
      "width": 535.7142857142858,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "rgb(218, 239, 255)",
          "borderColor": "#fff",
          "borderRadius": 5,
          "borderWidth": 2,
          "height": 30,
          "justifyContent": "center",
          "left": -1,
          "top": 10,
          "width": 30,
          "zIndex": 1,
        },
        undefined,
        Object {
          "left": 506.7142857142858,
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "backgroundColor": "rgb(218, 239, 255)",
          "borderTopRightRadius": 5,
          "marginVertical": 1,
          "paddingHorizontal": 7,
          "paddingVertical": 12,
        },
        Object {
          "borderTopLeftRadius": 5,
          "borderTopRightRadius": 0,
        },
      ]
    }
  >
    <Text
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontFamily": "TitilliumWeb-Regular",
            "fontSize": 17.5,
            "lineHeight": 22,
            "overflow": "visible",
          },
          Object {
            "color": "rgba(27, 106, 158, 0.85)",
            "fontFamily": "TitilliumWeb-SemiBold",
            "fontSize": 18.125,
            "lineHeight": 22,
          },
          Object {
            "color": "rgba(95, 95, 95, 0.87)",
            "fontFamily": "TitilliumWeb-SemiBold",
            "textAlign": "left",
          },
        ]
      }
    >
      title test
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "rgb(218, 239, 255)",
          "marginBottom": 1,
          "padding": 7,
        },
        Object {
          "borderBottomLeftRadius": 5,
          "borderBottomRightRadius": 5,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Text
        allowFontScaling={false}
        ellipsizeMode="tail"
        onPress={[Function]}
        style={
          Array [
            Object {
              "fontFamily": "TitilliumWeb-Regular",
              "fontSize": 17.5,
              "lineHeight": 22,
              "overflow": "visible",
            },
            Object {
              "color": "rgba(27, 106, 158, 0.85)",
              "fontSize": 17.5,
              "lineHeight": 21,
            },
            Array [
              Object {
                "color": "rgba(95, 95, 95, 0.87)",
                "paddingBottom": 2,
                "paddingTop": 4,
              },
              undefined,
            ],
          ]
        }
      >
        [no item subtitle]
         
      </Text>
    </View>
  </View>
</View>
`;
